### PR TITLE
add netping plugin for ilogtail

### PR DESCRIPTION
**Pull request by yonghua-sun**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,6 +7,7 @@ require (
 	github.com/StackExchange/wmi v0.0.0-20181212234831-e0a55b97c705 // indirect
 	github.com/VictoriaMetrics/VictoriaMetrics v1.58.0
 	github.com/VictoriaMetrics/metrics v1.17.2
+	github.com/agiledragon/gomonkey/v2 v2.4.0
```
Comment (EvanLjp) [inline go.mod:10]: please run `make check-license` to add new dependencies

File: plugins/input/netping/netping.go [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+	err = conn.Close()
+	if err != nil {
+		return 0, err
+	}
+
+	return time.Since(now), nil
+}
+
+// Init method would be triggered before working. In the example plugin, we set the initial
+// value of counter to 100. And we return 0 to use the default trigger interval.
```
Comment (EvanLjp) [inline plugins/input/netping/netping.go:96]: please remove these lines.

File: plugins/input/netping/netping.go [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+	context         ilogtail.Context
+	IntervalSeconds int          `json:"interval_seconds" comment:"the interval of ping/tcping, unit is second,must large than 30"`
+	ICMPConfigs     []ICMPConfig `json:"icmp" comment:"the icmping config list, example:  {\"src\" : \"${IP_ADDR}\",  \"target\" : \"${REMOTE_HOST}\", \"count\" : 3}"`
+	TCPConfigs      []TCPConfig  `json:"tcp" comment:"the tcping config list, example: {\"src\" : \"${IP_ADDR}\",  \"target\" : \"${REMOTE_HOST}\", \"port\" : ${PORT}, \"count\" : 3}"`
+}
+
+const MinIntervalSecods = 30 // min inteval should large than 30s
+
+// Get preferred outbound ip of this machine
+func getOutboudIP() (net.IP, error) {
```
Comment (EvanLjp) [inline plugins/input/netping/netping.go:70]: How about to use ` util.GetIPAddress()` ？

File: plugins/input/netping/netping.go [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+}
+
+func (m *NetPing) Description() string {
+	return "a icmp-ping/tcp-ping plugin for logtail"
+}
+
+// Collect is called every trigger interval to collect the metrics and send them to the collector.
+func (m *NetPing) Collect(collector ilogtail.Collector) error {
+
+	ch := make(chan *Result, 100)
```
Comment (EvanLjp) [inline plugins/input/netping/netping.go:144]: why not reuse the channel?

File: plugins/input/netping/netping.go [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+		}
+	}
+	if counter == 0 {
+		// nothing to do
+		return nil
+	}
+
+	for i := 0; i < counter; i++ {
+		result := <-ch
+		helper.AddMetric(collector, fmt.Sprintf("%s_total", result.Type), time.Now(), result.Label, float64(result.Total))
```
Comment (EvanLjp) [inline plugins/input/netping/netping.go:167]: time.Now() should get results as the first of the method.

File: plugins/input/netping/netping.go [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+}
+
+// NetPing struct implements the MetricInput interface.
+// The plugin has a counter field, which would be increment every 5 seconds.
+// And, the value of this counter will be sent as metrics data.
+type NetPing struct {
+	IcmpPrivileged  bool
+	HasConfig       bool
+	context         ilogtail.Context
+	IntervalSeconds int          `json:"interval_seconds" comment:"the interval of ping/tcping, unit is second,must large than 30"`
```
Comment (EvanLjp) [inline plugins/input/netping/netping.go:62]: comment should generate doc by the link:
https://github.com/alibaba/ilogtail/blob/main/docs/zh/guides/How-to-genernate-plugin-doc.md

File: plugins/input/netping/netping.go [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+	Port   int    `json:"port"`
+	Count  int    `json:"count"`
+}
+
+// NetPing struct implements the MetricInput interface.
+// The plugin has a counter field, which would be increment every 5 seconds.
+// And, the value of this counter will be sent as metrics data.
+type NetPing struct {
+	IcmpPrivileged  bool
+	HasConfig       bool
```
Comment (EvanLjp) [inline plugins/input/netping/netping.go:60]: HasConfig I don't find any position to use it.

File: plugins/input/netping/netping.go [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+	Target string `json:"target"`
+	Port   int    `json:"port"`
+	Count  int    `json:"count"`
+}
+
+// NetPing struct implements the MetricInput interface.
+// The plugin has a counter field, which would be increment every 5 seconds.
+// And, the value of this counter will be sent as metrics data.
+type NetPing struct {
+	IcmpPrivileged  bool
```
Comment (EvanLjp) [inline plugins/input/netping/netping.go:59]: If IcmpPrivileged is exported, should add comment tags.

**Review by EvanLjp — CHANGES_REQUESTED** [review]

File: plugins/input/netping/netping.go (hunk 4 above)
Comment (yonghua-sun) [reply plugins/input/netping/netping.go]: changed to used

File: plugins/input/netping/netping.go (hunk 3 above)
Comment (yonghua-sun) [reply plugins/input/netping/netping.go]: replaced

File: plugins/input/netping/netping.go (hunk 2 above)
Comment (yonghua-sun) [reply plugins/input/netping/netping.go]: removed

File: plugins/input/netping/netping.go (hunk 6 above)
Comment (yonghua-sun) [reply plugins/input/netping/netping.go:62]: added

File: plugins/input/netping/netping.go (hunk 7 above)
Comment (yonghua-sun) [reply plugins/input/netping/netping.go]: now will use in Collect method

File: plugins/input/netping/netping.go (hunk 8 above)
Comment (yonghua-sun) [reply plugins/input/netping/netping.go]: icmpPrivileged is used for ut , will not exported

**Pull request merged**